sys/net/link_layer/ng_slip/ng_slip.c

```diff
@@ -0,0 +1,293 @@
+/*
+ * Copyright (C) 2015 Martine Lenders <mlenders@inf.fu-berlin.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @{
+ *
+ * @file
+ */
+
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "kernel.h"
+#include "kernel_types.h"
+#include "msg.h"
+#include "net/ng_netbase.h"
+#include "periph/uart.h"
+#include "ringbuffer.h"
+#include "thread.h"
+#ifdef MODULE_NG_IPV6   /* TODO: remove guard when ng_ipv6 was merged */
+#include "net/ng_ipv6/hdr.h"
+#endif
+
+#include "net/ng_slip.h"
+
+#define ENABLE_DEBUG    (0)
+#include "debug.h"
+
+#if UART_NUMOF
+#define _SLIP_END         ('\xc0')
+#define _SLIP_ESC         ('\xdb')
+#define _SLIP_END_ESC     ('\xdc')
+#define _SLIP_ESC_ESC     ('\xdd')
+
+#define _SLIP_MSG_TYPE          (0xc1dc)    /* chosen randomly */
+#define _SLIP_STACK_SIZE        (KERNEL_CONF_STACKSIZE_DEFAULT)
+#define _SLIP_NAME              "SLIP"
+#define _SLIP_MSG_QUEUE_SIZE    (8U)
+
+static char _slip_stack[_SLIP_STACK_SIZE];
+
+static kernel_pid_t _slip_pid = KERNEL_PID_UNDEF;
+static uint32_t _bytes = 0;
+static uint8_t _esc = 0;
+static ringbuffer_t *_in_buf;
```

Comment:
I don't like this. You should put all these parameters into a device descriptor and use that as a parameter with the functions below. Otherwise it is not possible to run more than one slip interface at the time.